vector-im/element-android
# Refactor: split TimelineFragment into MessageComposerFragment and VoiceRecorderFragment

## Type of change

- [x] Technical: refactor.

## Content

Split `TimelineFragment` into 3 separate entities:
- TimelineFragment.
- MessageComposerFragment.
- VoiceRecorderFragment.

## Motivation and context

`TimelineFragment` is a large class that has many responsibilities: rendering timeline events, handling composer input, handling voice recording, etc. This is an attempt to separate those into separate components that can be worked on independently.

## Screenshots / GIFs

UI hasn't changed.

## Tests



- Go into any room.
- Try sending a message, see that the UI behaves as expected.
- Try also replies, quotes, attachments, editing events, etc.
- Try autocompletion.
- Try tapping on usernames in the timeline.
- Try voice recordings.
- In general, try anything that's related to either the message composer or voice recording.

## Tested devices

- OS version(s): 11, 12.

## Checklist
